sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/SortBasedAggregationIterator.scala

```diff
@@ -90,6 +91,24 @@ class SortBasedAggregationIterator(
   // compared to MutableRow (aggregation buffer) directly.
   private[this] val safeProj: Projection = FromUnsafeProjection(valueAttributes.map(_.dataType))
 
+  // Aggregation function which uses generic aggregation buffer object.
+  // @see [[TypedImperativeAggregate]] for more information
+  private val typedImperativeAggregates: Array[TypedImperativeAggregate[_]] = {
+    aggregateFunctions.collect {
+      case (ag: TypedImperativeAggregate[_]) => ag
+    }
+  }
+
+  // For TypedImperativeAggregate with generic aggregation buffer object, we need to call
+  // serializeAggregateBufferInPlace(...) explicitly to convert the aggregation buffer object
+  // to Spark Sql internally supported serializable storage format.
+  private def serializeTypedAggregateBuffer(aggregationBuffer: MutableRow): Unit = {
```

Comment:
Unused parameter `aggregationBuffer`. Or replace the following `sortBasedAggregationBuffer` to `aggregationBuffer`?